zcfan.c

```diff
@@ -40,7 +40,7 @@ static struct Rule rules[] = {
 };
 
 static struct timespec last_watchdog_ping = {0, 0};
-static const time_t watchdog_secs = 120;
+static int watchdog_secs = 120;
```

Comment:
This isn't correct, this must be kept as a `time_t` to make sure we match the architecture's signedness for later comparisons.

Reply:
Changed back to `time_t`.

For your information:

```
$ dpkg --print-architecture ; cc -E  -std=gnu99 -O2 -pedantic -Wall -Wextra -Wwrite-strings -Warray-bounds -Wconversion -Wstrict-prototypes -Werror  zcfan.c | grep time_t | grep typedef
i386
__extension__ typedef long int __time_t;
typedef __time_t time_t;
```

I'm not sure, I understand you correctly, but it seems to be, that on a standard i386 debian (only build-essential) is installed, gcc+glibc is still setting time_t to signed.  This is also explicitly mentioned here: https://sourceware.org/glibc/wiki/Y2038ProofnessDesign#time_t:~:text=time_t%20is%20a%2032%2Dbit%20signed%20Epoch%2Drelated%20count%20of%20seconds%2C%20and%20is%20Y2038%2Dincompatible%20because%20Y2038%20is%20the%20maximum%20positive%20value%20of%20time_t

As far as I remember, time_t was always signed, and when the whole Y2038 issues came up in 2000, one option was already to change it to unsigned and make it a Y2100 problem.  So I'm not aware of any architecture where time_t is unsigned (32 or 64 bit), but I might be wrong here.

Either case, I agree that it's best if this is time_t.  BTW, I tested compilation on both i386 and x86_64 now for the current PR, no warnings on either architecture.

Reply:
Hmm, I'm indeed mistaken from my memory from #8. It's almost certainly a matter of compiler warning behaviour rather than actual signedness changes. Anyway, better not to reply on knowing what the typedef is if possible.